### PR TITLE
Comparable messages

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we introduce the `compare_by` option. It allows users to specify the natural ordering strategy for messages, based on the values of their fields.

Example:
```proto
message LocalDate {
    option (compare_by) = {
        field: "year" field: "month" field: "day"
    };

    int32 year = 1;
    Month month = 2;
    uint32 day = 3;
}
```

Such declaration indicates that the value must be compared by the fields `year`, `month`, and `day` in that order.
Recursive comparisons are also supported:

```proto
message LocalDateTime {
    option (compare_by) = {
        field: "date" field: "time"
    };

    LocalDate date = 1;
    LocalTime time = 2;
}
```

Here, both `LocalDate` and `LocalTime` must be marked with the `(compare_by)` option as well as `LocalDateTime`. 

This feature is applicable to any Protobuf message.

**Note.** This PR merely provides a specification for the feature. The implementation, based on ProtoData, will come in a future PR. In order to avoid a circular dependency, the implementation will not declare a dependency on `:base`, and the new option must come from a transitive dependency via ProtoData.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: base/src/main/proto/spine/options.proto [hunk 1]
```diff
@@ -836,3 +844,42 @@ message IsOption {
...
     string java_type = 2;
 }
+
+// Defines the way to compare two messages of the same type to one another.
+//
+// Comparisons can be used to sort values.
+//
+// See the `(compare_by)` option.
+//
+message CompareOption {
```
Comment (armiol) [inline base/src/main/proto/spine/options.proto:854]: As discussed, let's call this `CompareByOption`, to leave `Compare...` unengaged for future use.

File: base/src/main/proto/spine/options.proto [hunk 2]
```diff
@@ -836,3 +844,42 @@ message IsOption {
...
+
+    // Field paths used for comparisons.
+    //
+    // The allowed field types are:
+    //  - any number type;
+    //  - `bool` (false is less than true);
+    //  - `string` (alphabetical order);
```
Comment (armiol) [inline base/src/main/proto/spine/options.proto:861]: "in the order of respective Unicode values"

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM with some discussion results.

**Pull request merged**